deeplake/core/vectorstore/deeplake_vectorstore.py

```diff
@@ -0,0 +1,213 @@
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        embeddings: Optional[Any] = None,
+    ) -> List[str]:
+        """Adding elements to deeplake vector store
+
+        Args:
+            texts (Iterable[str]): texts to add to deeplake vector store
+            metadatas (Optional[List[dict]], optional): List of metadatas.. Defaults to None.
+            ids (Optional[List[str]], optional): List of document IDs. Defaults to None.
+            embeddings (Optional[Any], optional): embedding of texts. Defaults to None.
+        Returns:
+            ids (List[str]): List of document IDs
+        """
+        elements = self._create_elements(ids, texts, metadatas, embeddings)
+        self._run_data_injestion(elements)
+        self.dataset.commit(allow_empty=True)
+        self.dataset.summary()
+        return ids
+
+    def query(
+        self,
+        query: Any[str, None] = None,
+        embedding: Any[float, None] = None,
+        k: int = 4,
+        distance_metric: str = "L2",
+        filter: Optional[Any[Dict[str, str], Callable, str]] = None,
+        exec_option: Optional[str] = None,
+    ):
+        view = self._attribute_based_filtering(filter)
+        if len(view) == 0:
+            return view
+
+        if self._embedding_function is None and embedding is None:
+            view, scores, indices = self._exact_text_search(view, query)
+        else:
+            emb = embedding or self._embedding_function.embed_query(
+                query
+            )  # type: ignore
+            query_emb = np.array(emb, dtype=np.float32)
+            embeddings = view.embedding.numpy(fetch_chunks=True)
+            indices, scores = vector_search.search(
+                query_embedding=query_emb,
+                embeddings=embeddings,
+                k=k,
+                distance_metric=distance_metric.lower(),
+                exec_option=exec_option or self._exec_option,
+                deeplake_dataset=self.dataset,
+            )
+            return (view, indices, scores)
+
+    def delete(
+        self,
+        ids: Any[List[str], None] = None,
+        filter: Any[Dict[str, str], None] = None,
+        delete_all: Any[bool, None] = None,
+    ) -> bool:
+        """Delete the entities in the dataset
+        Args:
+            ids (Optional[List[str]], optional): The document_ids to delete.
+                Defaults to None.
+            filter (Optional[Dict[str, str]], optional): The filter to delete by.
+                Defaults to None.
+            delete_all (Optional[bool], optional): Whether to drop the dataset.
+                Defaults to None.
+        """
+        if delete_all:
+            self.dataset.delete(large_ok=True)
+            return True
+
+        view = None
+        if ids:
+            view = self.dataset.filter(lambda x: x["ids"].data()["value"] in ids)
+            ids = list(view.sample_indices)
+
+        if filter:
```

Comment:
How TQL is handling filters? this should not be here if TQL is running it.

Reply:
Not getting what is the problem with TQL. TQL is used to search the data, and sends back only scores in numpy format and indices, and does not have anything to do with the dataset or filters.